xc/device.ml

```diff
@@ -2089,6 +2105,9 @@ module Backend = struct
     | Profile.Qemu_upstream        -> (module Qemu_upstream        : Intf)
 
   let of_domid x = of_profile (Profile.of_domid x)
+
+  let init() =
+    Profile.all |> List.iter (fun p -> let module Q = (val of_profile p) in Q.Dm.Event.init() )
```

Comment:
@fillzero this line looked like a good idea at the time, but I think it may be creating 2 threads, one for the profile Qemu_upstream_compat, and another for the profile Qemu_upstream.

I think it should be replaced by a single direct call to
```
Qemu_upstream.Dm.Event.init()
```
so that we are sure that a single thread is created to handle the QMP events.